### ISSUE TITLE
order in the rules 

### ISSUE THREAD

**Issue by JihedC** [KoesGroup/Snakemake_ChIPseq#8]
When running the snakefile with multiple core, it seems that the indexing of the bam file is done later than the rules using it.
`'results/mapped/ChIP1_L1.sorted.rmdup.bam' does not appear to have an index. You MUST index the file first!`
```
'results/mapped/ChIP1_L1.sorted.rmdup.bam' does not appear to have an index. You MUST index the file first!
    Error in rule bamcompare:
        jobid: 3
        output: results/bamcompare/log2_ChIP1_ChIP2_L1.bamcompare.bw

RuleException:
CalledProcessError in line 249 of /Users/Jihed/Desktop/DMC1_ChIPseq/Snakefile:
Command ' set -euo pipefail;  bamCompare -b1 results/mapped/ChIP1_L1.sorted.rmdup.bam -b2 results/mapped/ChIP2_L1.sorted.rmdup.bam -o results/bamcompare/log2_ChIP1_ChIP2_L1.bamcompare.bw ' returned non-zero exit status 1
  File "/Users/Jihed/Desktop/DMC1_ChIPseq/Snakefile", line 249, in __rule_bamcompare
  File "/anaconda3/envs/bigwig/lib/python3.5/concurrent/futures/thread.py", line 55, in run```

The rule producing the bam index file is `rule bam_index`, take sorted bam files as input. The output is required by the `rule_all`, so the file are produced at some point and it appears in the DAG. However I can not find the files `*.bai` in the output folder.

**Comment by mgalland** [comment]
On which branch? On the bed branch?

**Comment by JihedC** [comment]
Yeah sorry I forgot to tell, it's the bed branch

**Comment by JihedC** [comment]
Yes I think so.
But I have made a mistake :(,  instead of creating a new branch to solve the issue with the environments and the `--use-conda`, I have solved it in the same branch [commit ed43613](https://github.com/KoesGroup/Snakemake_ChIPseq/pull/7/commits/ed436132f87dacbdbcda64f94d7f47b0fa06c462).

On my computer it runs when I use `snakemake --use-conda`, the error on the missing bam index file for the rules using it is not there anymore.

There is a little typo on the [line 214](https://github.com/KoesGroup/Snakemake_ChIPseq/blob/5da94d8c0971d86f0a98e15d0b85b30885dbfaa9/Snakefile#L214), it should be `{output.bai}` and not `{output.bam}` but it still works fine.
